api/graphql/schema.graphql

```diff
@@ -532,6 +544,24 @@ type Role {
   displayName: LocalizedString @rename(attribute: "display_name")
   description: LocalizedString
   isTeamBased: Boolean @rename(attribute: "is_team_based")
+  # teams and roles: built-in
+  users: [UserPublicProfile!] @morphToMany
+  permissions: [Permission!] @belongsToMany
+  # teams and roles: custom pivot model
+  roleAssignments: [RoleAssignment!] @hasMany
+}
+
+# This model represents the polymorphic role_user table and allows querying the team relationship
+type RoleAssignment {
+  id: ID!
+  role: Role @belongsTo
+  team: Team @belongsTo
+  user: UserPublicProfile @morphTo
+}
+
+# stub for Laratrust
+type Permission {
+  id: ID!
 }
```

Comment:
I don't think we need to add this to schema yet if we have no use for it.

Reply:
Yeah, let's not add things we don't need yet.  Removed in [remove extra relationships](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5749/commits/69500c02baa6a090cdde1213656bb8672c78251c).